File: app/src/main/java/com/kazakago/cleanarchitecture/presentation/presenter/activity/MainActivityViewModel.java
package com.kazakago.cleanarchitecture.presentation.presenter.activity;

import android.content.Context;
import android.view.MenuItem;

import com.kazakago.cleanarchitecture.presentation.listener.presenter.activity.MainActivityViewModelListener;

/**
 * Main Activity ViewModel
 *
 * @author Kensuke
 */
public class MainActivityViewModel {

    private Context context;
    private MainActivityViewModelListener listener;

    public MainActivityViewModel(Context context, MainActivityViewModelListener listener) {
        this.context = context;
        this.listener = listener;
    }

    public void onClickAboutMenu(MenuItem item) {
        listener.toAboutActivity();
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/presentation/presenter/activity/MainActivityViewModel.kt
package com.kazakago.cleanarchitecture.presentation.presenter.activity

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LifecycleObserver
import com.kazakago.cleanarchitecture.presentation.listener.presenter.activity.MainActivityViewModelListener

class MainActivityViewModel(application: Application) : AndroidViewModel(application), LifecycleObserver {

    var listener: MainActivityViewModelListener? = null

    fun onClickAboutMenu() {
        listener?.toAboutActivity()
    }

}
